srtcore/srt_compat.h

```diff
@@ -20,49 +20,6 @@ written by
 #include <stddef.h>
 #include <time.h>
 
-#ifndef SRT_API
-#ifdef _WIN32
-   #ifndef __MINGW32__
-      #ifdef SRT_DYNAMIC
-         #ifdef SRT_EXPORTS
-            #define SRT_API __declspec(dllexport)
-         #else
-            #define SRT_API __declspec(dllimport)
-         #endif
-      #else
-         #define SRT_API
-      #endif
-   #else
-      #define SRT_API
-   #endif
-#else
-   #define SRT_API __attribute__ ((visibility("default")))
-#endif
-#endif
-
-#ifdef _WIN32
-   // https://msdn.microsoft.com/en-us/library/tcxf1dw6.aspx
-   // printf() Format for ssize_t
-   #if !defined(PRIzd)
-      #define PRIzd "Id"
-   #endif
-   // printf() Format for size_t
-   #if !defined(PRIzu)
-      #define PRIzu "Iu"
-   #endif
-#else
-   // http://www.gnu.org/software/libc/manual/html_node/Integer-Conversions.html
-   // printf() Format for ssize_t
-   #if !defined(PRIzd)
-      #define PRIzd "zd"
-   #endif
-   // printf() Format for size_t
-   #if !defined(PRIzu)
-      #define PRIzu "zu"
-   #endif
-#endif
```

Comment:
These are removed. Not sure if `PRIzd` and `PRIzu` are needed anywhere though.
@jlsantiago0 you added them in PR #336 almost six years ago. Do you happen to know/remember why?

Reply:
About 3 years ago I took the effort of a complete removal of any printf-formatting things from SRT. This PRI* stuff was one of the reasons.

They should not be needed anywhere, and if they are, they should be removed. There's also an appriate annotation at the LOGF/HLOGF macros that they are deprecated.

Reply:
The printf versions weren't exactly abandoned due to performance, rather due to that printf formatting was causing problems, incompatibilities, unexpected warnings and portability issues. One of the most annoying things - and the reason why this PRI* stuff exists - was that the formatting tags in printf are tied to the builtin types only, while SRT makes wide use of types like `size_t` and `int64_t`, which's definition is platform-dependent. Only `size_t` got finally the `z` printf tag to support it, but that's also not implemented correctly in every compiler.

So everntually getting rid of the printf tag use was the most sensible choice. Performace is another matter - if you remember our last discussion about the {fmt} library, it turns out that iostreams with formatting are pretty slow (another thing is that we use specific formatting parameters rarely). Of course it would be nice to provide some better performant formatting library, but not such that would reuse the printf formatting and all its problems, and the best would be something that could be used as a drop-in replacement without changing the existing log instructions.